### PR TITLE
fix(ui): modal component/service/panel, popover fix. Closes #269

### LINKED ISSUE #269: Modal component closing issues
**I'm submitting a ...**  (check one with "x")

* [x] bug report
* [ ] feature request

**Current behavior:**

When trying to show modal after previous one is closed, both are shown

**Steps to reproduce:**

See [demo gif](https://take.ms/s5tDn)

### PR DESCRIPTION
**Pull request by 32penkin**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/react-native-ui-kitten/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/react-native-ui-kitten/blob/master/DEV_DOCS.md) guide.

 #### Short description of what this resolves:

### CODE REVIEW

File: src/framework/theme/component/modal/modalPanel.component.tsx [hunk 1]
```diff
@@ -4,53 +4,64 @@ import {
...
 import { Modal } from '../../../ui/modal/modal.component';
+import { ModalComponentCloseProps } from '@kitten/theme';
 
 export interface ModalPanelProps {
   children: React.ReactElement<any> | React.ReactElement<any>[];
 }
 
 interface ModalPanelState {
-  dialogComponents: Map<string, React.ReactElement<any>>;
+  dialogComponents: Map<string, React.ReactElement<ModalComponentCloseProps>>;
```
Comment (artyorsh) [inline src/framework/theme/component/modal/modalPanel.component.tsx]: components

File: src/framework/theme/component/modal/modalPanel.component.tsx [hunk 2]
```diff
@@ -4,53 +4,64 @@ import {
...
   backdropValues: Map<string, boolean>;
```
Comment (artyorsh) [inline src/framework/theme/component/modal/modalPanel.component.tsx]: backdrops

File: src/framework/theme/component/modal/modalPanel.spec.tsx [hunk 3]
```diff
@@ -28,6 +29,8 @@ describe('@modal panel checks', () => {
 
   class ModalPanelTest extends React.Component<HooksProps> {
 
+    private modalId: string = '';
+
     componentDidMount(): void {
```
Comment (artyorsh) [inline src/framework/theme/component/modal/modalPanel.spec.tsx]: prefer using access modifiers

File: src/framework/theme/component/modal/modalPanel.spec.tsx [hunk 4]
```diff
@@ -28,6 +29,8 @@ describe('@modal panel checks', () => {
...
       this.props.componentDidMount && this.props.componentDidMount();
```
Comment (artyorsh) [inline src/framework/theme/component/modal/modalPanel.spec.tsx]: prefer using control flow statements like `if`

File: src/playground/src/ui/screen/popover.component.tsx [hunk 5]
```diff
@@ -73,7 +73,8 @@ class Popover extends React.Component<Props, State> {
             style={component}
             placement={`${PLACEMENT} start`}
             visible={this.state.startVisible}
-            content={this.createPopoverContentElement('❤️')}>
+            content={this.createPopoverContentElement('❤️')}
```
Comment (artyorsh) [inline src/playground/src/ui/screen/popover.component.tsx]: You have lost the heart :(

File: src/framework/ui/popover/popover.component.tsx [hunk 6]
```diff
@@ -41,7 +42,7 @@ interface State {
   layout: MeasureResult | undefined;
 }
 
-export type Props = PopoverProps & StyledComponentProps & PopoverViewProps & ViewProps;
+export type Props = PopoverProps & ModalComponentCloseProps & StyledComponentProps & PopoverViewProps & ViewProps;
 
 const TAG_CHILD: number = 0;
```
Comment (malashkevich) [inline src/framework/ui/popover/popover.component.tsx]: simplify this

File: src/framework/theme/component/modal/modalPanel.component.tsx (hunk 1 above)
Comment (32penkin) [reply src/framework/theme/component/modal/modalPanel.component.tsx]: Done.

File: src/framework/theme/component/modal/modalPanel.component.tsx (hunk 2 above)
Comment (32penkin) [reply src/framework/theme/component/modal/modalPanel.component.tsx]: Done.

File: src/framework/theme/component/modal/modalPanel.spec.tsx (hunk 3 above)
Comment (32penkin) [reply src/framework/theme/component/modal/modalPanel.spec.tsx]: Done.

File: src/framework/theme/component/modal/modalPanel.spec.tsx (hunk 4 above)
Comment (32penkin) [reply src/framework/theme/component/modal/modalPanel.spec.tsx]: Done.

File: src/playground/src/ui/screen/popover.component.tsx (hunk 5 above)
Comment (32penkin) [reply src/playground/src/ui/screen/popover.component.tsx]: It was just an extra heart. Why do you need one more than mine?

**Pull request merged**